handlers/meta/whatsapp/templates.go

```diff
@@ -47,31 +47,23 @@ func GetTemplating(msg courier.MsgOut) (*MsgTemplating, error) {
 	return metadata.Templating, nil
 }
 
-func GetTemplatePayload(templating MsgTemplating) *Template {
-	template := &Template{Name: templating.Template.Name, Language: &Language{Policy: "deterministic", Code: templating.Language}}
+func GetTemplatePayload(templating *MsgTemplating) *Template {
+	template := &Template{
+		Name:       templating.Template.Name,
+		Language:   &Language{Policy: "deterministic", Code: templating.Language},
+		Components: []*Component{},
+	}
 
 	compKeys := maps.Keys(templating.Params)
 	sort.Strings(compKeys) // so that final component order is deterministic
 
 	for _, k := range compKeys {
 		v := templating.Params[k]
+		var component *Component
 
-		if strings.HasPrefix(k, "button.") {
-
-			for _, p := range v {
-				if p.Type == "url" {
-					component := &Component{Type: "button", Index: strings.TrimPrefix(k, "button."), SubType: "quick_reply"}
-					component.Params = append(component.Params, &Param{Type: "text", Text: p.Value})
-					template.Components = append(template.Components, component)
-				} else {
-					component := &Component{Type: "button", Index: strings.TrimPrefix(k, "button."), SubType: "quick_reply"}
-					component.Params = append(component.Params, &Param{Type: "payload", Payload: p.Value})
-					template.Components = append(template.Components, component)
```

Comment:
isn't this adding a button component for every param in a button ?

Reply:
I see